specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json

```diff
@@ -0,0 +1,646 @@
+        }
+      }
+    },
+    "Tags": {
+      "type": "object",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Tags field of the resource."
+        }
+      },
+      "description": "Tags field of the resource."
+    },
+    "SapMonitorListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SapMonitor"
+          },
+          "description": "The list of SAP monitors."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of SAP monitors."
+        }
+      },
+      "description": "The response from the List SAP monitors operation."
+    },
+    "SapMonitor": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SapMonitorProperties",
+          "description": "SAP monitor properties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "SAP monitor info on Azure (ARM properties and SAP monitor properties)"
+    },
+    "SapMonitorProperties": {
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "enum": [
+            "Accepted",
+            "Creating",
+            "Updating",
+            "Failed",
+            "Succeeded",
+            "Deleting",
+            "Migrating"
+          ],
+          "x-ms-enum": {
+            "name": "HanaProvisioningStatesEnum",
+            "modelAsString": true
+          },
+          "description": "State of provisioning of the HanaInstance"
+        },
+        "managedResourceGroupName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the resource group the SAP Monitor resources get deployed into."
+        },
+        "logAnalyticsWorkspaceArmId": {
+          "type": "string",
+          "description": "The ARM ID of the Log Analytics Workspace that is used for monitoring"
+        },
+        "enableCustomerAnalytics": {
+          "type": "boolean",
+          "description": "The value indicating whether to send analytics to Microsoft"
+        },
+        "logAnalyticsWorkspaceId": {
+          "type": "string",
+          "description": "The workspace ID of the log analytics workspace to be used for monitoring"
+        },
+        "logAnalyticsWorkspaceSharedKey": {
+          "type": "string",
+          "description": "The shared key of the log analytics workspace that is used for monitoring"
+        },
+        "sapMonitorCollectorVersion": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The version of the payload running in the Collector VM"
+        },
+        "providerInstances": {
```

Comment:
should not include child types in the parent - pagination will  be a problem, RBAC will be a aproblem. USer can do a collection GET to get all providerInstances under a provider.

Reply:
providerInstances is a child resource to sapMonitor.
I want to model it like how securityRules is a child resource to networkSecurityGroups
What is the guidance on achieving that?

Reply:
user can do explicit GET on the child resource

Reply:
We can't mimic what networkSecurityGroups and securityRules do?

Reply:
Changes fix this issue